tests/test_tokenization_fast.py

```diff
@@ -90,3 +91,21 @@ def test_training_new_tokenizer_with_special_tokens_change(self):
                     # is restored
                     shutil.rmtree(self.tmpdirname)
                     self.tmpdirname = tmpdirname_orig
+
+
+@require_tokenizers
+class ReduceMutableBorrowTests(unittest.TestCase):
+    def test_async_share_tokenizer(self):
```

Comment:
Great test! I could have been a very small suggestion: wouldn't we like to leave a comment to explain why we need to do this test? (maybe mentioning the issue #https://github.com/huggingface/tokenizers/issues/537)